File: FE/WEB/app/models/channel-pricing.js
import Model, { attr, belongsTo } from '@ember-data/model';

/**
 * Channel Pricing Model
 * Tariffe e abbonamenti specifici per ciascun canale
 * Range di prezzi (from/to) per ogni tipologia di servizio
 */
export default class ChannelPricingModel extends Model {
  // ============================================================
  // RELATIONSHIPS
  // ============================================================

  /**
   * FK to channel (1:1 relationship)
   */
  @belongsTo('channel', { async: true, inverse: 'pricing' }) channel;

  // ============================================================
  // ATTRIBUTES - Price Ranges
  // ============================================================

  /**
   * Abbonamento mensile
   */
  @attr('number') monthlySubscriptionFrom;
  @attr('number') monthlySubscriptionTo;

  /**
   * Vendita foto
   */
  @attr('number') photoSaleFrom;
  @attr('number') photoSaleTo;

  /**
   * Vendita video
   */
  @attr('number') videoSaleFrom;
  @attr('number') videoSaleTo;

  /**
   * Live public
   */
  @attr('number') livePublicFrom;
  @attr('number') livePublicTo;

  /**
   * Live private
   */
  @attr('number') livePrivateFrom;
  @attr('number') livePrivateTo;

  /**
   * Vendita abbigliamento
   */
  @attr('number') clothingSalesFrom;
  @attr('number') clothingSalesTo;

  /**
   * Contenuti extra
   */
  @attr('number') extraContentFrom;
  @attr('number') extraContentTo;

  /**
   * Note aggiuntive
   */
  @attr('string') note;

  /**
   * Timestamps
   */
  @attr('date-utc') createdAt;
  @attr('date-utc') updatedAt;

  // ============================================================
  // HELPER METHODS
  // ============================================================

  /**
   * Formatta un range di prezzi
   * @param {number} from - Prezzo minimo
   * @param {number} to - Prezzo massimo
   * @returns {string|null} Range formattato o null
   */
  priceRange(from, to) {
    if (!from && !to) return null;
    if (from === to) return `€${from}`;
    if (!to) return `da €${from}`;
    if (!from) return `fino a €${to}`;
    return `€${from} - €${to}`;
  }

  // ============================================================
  // COMPUTED PROPERTIES
  // ============================================================

  /**
   * Range abbonamento mensile formattato
   */
  get monthlySubscriptionRange() {
    return this.priceRange(
      this.monthlySubscriptionFrom,
      this.monthlySubscriptionTo
    );
  }

  /**
   * Range vendita foto formattato
   */
  get photoSaleRange() {
    return this.priceRange(this.photoSaleFrom, this.photoSaleTo);
  }

  /**
   * Range vendita video formattato
   */
  get videoSaleRange() {
    return this.priceRange(this.videoSaleFrom, this.videoSaleTo);
  }

  /**
   * Range live public formattato
   */
  get livePublicRange() {
    return this.priceRange(this.livePublicFrom, this.livePublicTo);
  }

  /**
   * Range live private formattato
   */
  get livePrivateRange() {
    return this.priceRange(this.livePrivateFrom, this.livePrivateTo);
  }

  /**
   * Range vendita abbigliamento formattato
   */
  get clothingSalesRange() {
    return this.priceRange(this.clothingSalesFrom, this.clothingSalesTo);
  }

  /**
   * Range contenuti extra formattato
   */
  get extraContentRange() {
    return this.priceRange(this.extraContentFrom, this.extraContentTo);
  }

  /**
   * Check se ha almeno un prezzo impostato
   * @returns {boolean}
   */
  get hasPricing() {
    return !!(
      this.monthlySubscriptionFrom ||
      this.monthlySubscriptionTo ||
      this.photoSaleFrom ||
      this.photoSaleTo ||
      this.videoSaleFrom ||
      this.videoSaleTo ||
      this.livePublicFrom ||
      this.livePublicTo ||
      this.livePrivateFrom ||
      this.livePrivateTo ||
      this.clothingSalesFrom ||
      this.clothingSalesTo ||
      this.extraContentFrom ||
      this.extraContentTo
    );
  }
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/ChannelPricing.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    /// <summary>
    /// Pricing tiers and subscription fees for a channel
    /// 1:1 relationship with Channel
    /// </summary>
    [Resource]
    [Table("ChannelPricing")]
    public class ChannelPricing : Identifiable<int>
    {
        [Attr]
        [Required]
        public string ChannelId { get; set; }

        // Monthly Subscription Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? MonthlySubscriptionFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? MonthlySubscriptionTo { get; set; }

        // Photo Sale Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? PhotoSaleFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? PhotoSaleTo { get; set; }

        // Video Sale Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? VideoSaleFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? VideoSaleTo { get; set; }

        // Live Public Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? LivePublicFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? LivePublicTo { get; set; }

        // Live Private Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? LivePrivateFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? LivePrivateTo { get; set; }

        // Clothing Sales Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? ClothingSalesFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? ClothingSalesTo { get; set; }

        // Extra Content Range
        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? ExtraContentFrom { get; set; }

        [Attr]
        [Column(TypeName = "decimal(18,2)")]
        public decimal? ExtraContentTo { get; set; }

        [Attr]
        public string Note { get; set; }

        // Timestamps
        [Attr]
        public DateTime CreatedAt { get; set; }

        [Attr]
        public DateTime UpdatedAt { get; set; }

        // Navigation Properties
        [HasOne]
        public virtual Channel Channel { get; set; }
    }
}
